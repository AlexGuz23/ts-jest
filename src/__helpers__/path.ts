import { resolve, relative } from 'path'

export const ROOT = resolve(__dirname, '..', '..')

export function relativeToRoot(path: string) {
  return relative(ROOT, path)
}
