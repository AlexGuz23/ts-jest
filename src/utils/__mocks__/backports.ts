import * as _backports from '../backports';

export const backportJestConfig: typeof _backports['backportJestConfig'] = val => ({
  ...(val as any),
});
