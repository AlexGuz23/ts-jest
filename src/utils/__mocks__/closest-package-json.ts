import _closestPackageJson from '../closest-package-json';
import { TPackageJson } from '../../types';

// tslint:disable-next-line:variable-name
export const __byPath: Record<string, TPackageJson> = {};
// tslint:disable-next-line:variable-name
export const __default: TPackageJson = { name: 'mock' };

const closestPackageJson: typeof _closestPackageJson = (
  fromPath: string,
  asString: boolean = false,
) => {
  const res = fromPath in __byPath ? __byPath[fromPath] : __default;
  return (asString ? JSON.stringify(res) : res) as any;
};

export default closestPackageJson;
