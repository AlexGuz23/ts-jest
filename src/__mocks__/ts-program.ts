import * as fakers from '../__helpers__/fakers'
import { TsJestConfig, TsJestProgram } from '../types'
import { ParsedCommandLine } from 'typescript'

// tslint:disable-next-line:variable-name
export let __tsConfig: any = {}

export default class TsProgramMock implements TsJestProgram {
  get parsedConfig(): ParsedCommandLine {
    return __tsConfig
  }

  constructor(
    public rootDir: string = fakers.filePath(''),
    public tsJestConfig: TsJestConfig = fakers.tsJestConfig(),
  ) {}

  transpileModule(_: string, source: string) {
    return source
  }
}
