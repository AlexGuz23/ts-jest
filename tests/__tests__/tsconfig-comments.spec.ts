import { MockedPath } from '../__mocks__/path';
jest.mock('path');
import * as fs from 'fs';
import getTSConfig from '../../dist/utils/get-ts-config';
import * as path from 'path';
import cfg from '../__helpers__/jest-config';

describe('parse tsconfig with comments', () => {
  const configFile1 = './tests/tsconfig-test/allows-comments.json';
  const configFile2 = './tests/tsconfig-test/allows-comments2.json';

  beforeEach(() => {
    // Set up some mocked out file info before each test
    ((path as any) as MockedPath).__setBaseDir('./tests/tsconfig-test');
    getTSConfig.cache.clear();
  });

  it('the two config files should exist', () => {
    expect(fs.existsSync(configFile1)).toBe(true);
    expect(fs.existsSync(configFile2)).toBe(true);
  });

  it('the two config files should have comments', () => {
    // We test for comments by trying to use JSON.parse
    // which should fail.

    expect(() => {
      JSON.parse(fs.readFileSync(configFile1, 'utf8'));
    }).toThrow();

    expect(() => {
      JSON.parse(fs.readFileSync(configFile2, 'utf8'));
    }).toThrow();
  });

  describe('new behaviour (tsConfigFile & tsConfig)', () => {
    // allows-comments.json does not contain a "pretty" field,
    // while allows-comments2.json does.
    // allow-comments.json extends allow-comments2.json
    it('should correctly read allow-comments.json', () => {
      const config = getTSConfig(
        cfg.tsconfigTest({ tsConfigFile: 'allows-comments.json' }),
      );
      expect(config).toMatchSnapshot();
    });
    it('should correctly read allow-comments2.json', () => {
      const config = getTSConfig(
        cfg.tsconfigTest({ tsConfigFile: 'allows-comments2.json' }),
      );
      expect(config).toMatchSnapshot();
    });
  });
});
