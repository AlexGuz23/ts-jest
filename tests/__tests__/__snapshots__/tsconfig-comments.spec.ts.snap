// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse tsconfig with comments new behaviour (tsConfigFile & tsConfig) should correctly read allow-comments.json 1`] = `
Object {
  "configFilePath": undefined,
  "declaration": false,
  "declarationMap": false,
  "emitDeclarationOnly": false,
  "inlineSourceMap": true,
  "inlineSources": true,
  "jsx": 2,
  "module": 1,
  "pretty": false,
}
`;

exports[`parse tsconfig with comments new behaviour (tsConfigFile & tsConfig) should correctly read allow-comments2.json 1`] = `
Object {
  "configFilePath": undefined,
  "declaration": false,
  "declarationMap": false,
  "emitDeclarationOnly": false,
  "inlineSourceMap": true,
  "inlineSources": true,
  "jsx": 2,
  "module": 1,
  "pretty": false,
}
`;
